examples/medplum-chat-demo/src/App.tsx

```diff
@@ -0,0 +1,98 @@
+import { formatSearchQuery, getReferenceString, Operator, ProfileResource } from '@medplum/core';
+import { AppShell, Loading, Logo, NotificationIcon, useMedplum, useMedplumProfile } from '@medplum/react';
+import { IconClipboardCheck, IconFileImport, IconMail, IconMessage, IconMessage2Bolt } from '@tabler/icons-react';
+import { Suspense } from 'react';
+import { Route, Routes, useNavigate } from 'react-router-dom';
+import { CommunicationPage } from './pages/CommunicationPage';
+import { LandingPage } from './pages/LandingPage';
+import { PatientPage } from './pages/PatientPage';
+import { ResourcePage } from './pages/ResourcePage';
+import { SearchPage } from './pages/SearchPage';
+import { SignInPage } from './pages/SignInPage';
+import { UploadDataPage } from './pages/UploadDataPage';
+
+export function App(): JSX.Element | null {
+  const medplum = useMedplum();
+  const profile = useMedplumProfile();
+  const navigate = useNavigate();
+
+  const profileReference = (profile && getReferenceString(profile)) as string;
+  const userLinks = [
+    { icon: <IconMessage />, label: 'All Threads', href: '/Communication?part-of:missing=true&status:not=completed' },
+  ];
+
+  const myThreadsQuery = formatSearchQuery({
+    resourceType: 'Communication',
+    filters: [
+      { code: 'part-of:missing', operator: Operator.EQUALS, value: 'true' },
+      { code: 'recipient', operator: Operator.EQUALS, value: profileReference },
+      { code: 'status:not', operator: Operator.EQUALS, value: 'completed' },
+    ],
+  });
+
+  userLinks.push({ icon: <IconMessage2Bolt />, label: 'My Threads', href: `/Communication${myThreadsQuery}` });
+
+  if (medplum.isLoading()) {
+    return null;
+  }
+
+  return (
+    <AppShell
+      logo={<Logo size={24} />}
+      menus={[
+        {
+          title: 'My Links',
+          links: userLinks,
+        },
+        {
+          title: 'Upload Data',
+          links: [{ icon: <IconFileImport />, label: 'Upload Example Data', href: 'upload/example' }],
+        },
+      ]}
+      notifications={
+        profile && (
+          <>
+            <NotificationIcon
```

Comment:
This is getting large. Break out each of these into a separate component in this file (`MessageNotification`, `TaskNotification`)